anndata/_core/views.py

```diff
@@ -295,8 +295,8 @@ def __copy__(self) -> AwkArray:
             """
             array = self
             # makes a shallow copy and removes the reference to the original AnnData object
-            array = ak.with_parameter(self, _PARAM_NAME, None)
-            array = ak.with_parameter(array, "__list__", None)
+            array = ak.with_parameter(array, _PARAM_NAME, None)
+            array = ak.with_parameter(array, "__record__", None)
```

Comment:
This should use `with_name`, so that we can traverse into the layout to find the record-node
```suggestion
            array = ak.with_name(array, None)
```

Reply:
I imagine that after our dicsussion @grst could not get that working (see https://github.com/scverse/anndata/pull/1040#issuecomment-1647878246)

Is there a reason traversing is better than what we have (assuming @grst could not get the `with_name` working)?  We only ever deal with this at a "top level."